### PR TITLE
migration to PTL 2.0 for spellmapper model

### PR DESCRIPTION
**Pull request by bene-ges**
# What does this PR do ?

migration to PTL 2.0 for spellmapper model, same as in #7597

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by bene-ges** [comment]
@athitten this change is similar to yours, could you review?

**Comment by bene-ges** [comment]
@titu1994, can you review or assign somebody?

File: nemo/collections/nlp/models/spellchecking_asr_customization/spellchecking_model.py [hunk 1]
```diff
@@ -272,14 +272,15 @@ def validation_step(self, batch, batch_idx):
...
         return {'val_loss': val_loss}
 
-    def validation_epoch_end(self, outputs):
+    def on_validation_epoch_end(self):
         """
         Called at the end of validation to aggregate outputs.
         :param outputs: list of individual outputs of each validation step.
         """
-        avg_loss = torch.stack([x['val_loss'] for x in outputs]).mean()
+        avg_loss = torch.stack([x['val_loss'] for x in self.validation_step_outputs]).mean()
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply nemo/collections/nlp/models/spellchecking_asr_customization/spellchecking_model.py]: @athitten , I fixed it in new commit

[the remark this answers is not in the document]
Comment (bene-ges) [reply nemo/collections/nlp/models/spellchecking_asr_customization/spellchecking_model.py]: I checked `/NeMo/examples/nlp/spellchecking_asr_customization/run_training.sh` - it works ok.

File: nemo/collections/nlp/models/spellchecking_asr_customization/spellchecking_model.py [hunk 2]
```diff
@@ -288,8 +298,8 @@
         logging.info("Total tag accuracy: " + str(tag_accuracy))
         logging.info(tag_report)
 
-        self.log('val_loss', avg_loss, prog_bar=True)
-        self.log('tag accuracy', tag_accuracy)
+        self.log(f"{split}_loss", avg_loss, prog_bar=True)
```
[the remark this answers is not in the document]
Comment (bene-ges) [reply nemo/collections/nlp/models/spellchecking_asr_customization/spellchecking_model.py:301]: fixed

**Comment by bene-ges** [comment]
@titu1994, jenkins passed

**Comment by bene-ges** [comment]
@athitten, @ericharper what about this pr? All checks passed

**Review by athitten — APPROVED** [review]
LGTM, thanks!

**Pull request merged**